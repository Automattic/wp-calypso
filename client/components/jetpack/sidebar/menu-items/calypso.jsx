/**
 * External dependencies
 */
import React from 'react';

/**
 * Internal dependencies
 */
import JetpackSidebarMenuItems from '.';

export default ( { path, expandSection } ) => {
	return (
		<JetpackSidebarMenuItems
			path={ path }
			showIcons={ false }
			tracksEventNames={ {
				activityClicked: 'calypso_mysites_jetpack_sidebar_activity_clicked',
				backupClicked: 'calypso_mysites_jetpack_sidebar_backup_clicked',
				scanClicked: 'calypso_mysites_jetpack_sidebar_scan_clicked',
			} }
			expandSection={ expandSection }
		/>
	);
};
