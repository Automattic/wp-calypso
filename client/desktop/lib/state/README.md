State
==========

Provides user state functions, initially for logged in state, but easily
extended to other cases at they rise

