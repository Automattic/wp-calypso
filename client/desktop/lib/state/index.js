/**
 * Internal dependencies
 */
const keychain = require( 'desktop/lib/keychain' );

/**
 * Module variables
 */

let state = false;

function State() {
	this.loggedIn = false;
	this.user = false;
}

State.prototype.isLoggedIn = function () {
	return this.loggedIn;
};

State.prototype.login = async function ( { user, token } ) {
	this.loggedIn = true;

	if ( user && token ) {
		this.user = {
			id: user.data.ID,
			token: `${ token }`,
		};
		keychain.setUserInfo( this.user );
	}
};

State.prototype.getUser = function () {
	if ( ! this.user && this.loggedIn ) {
		this.user = keychain.getUserInfo();
	}
	return this.user;
};

State.prototype.logout = function () {
	this.loggedIn = false;
	this.user = false;
	keychain.clear();
};

State.prototype.setLogPath = function ( path ) {
	this.logPath = path;
};

State.prototype.getLogPath = function () {
	return this.logPath;
};

if ( ! state ) {
	state = new State();
}

module.exports = state;
