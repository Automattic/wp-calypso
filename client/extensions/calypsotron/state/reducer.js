/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import productList from './product-list/reducer';

export default combineReducers( {
	productList,
} );
