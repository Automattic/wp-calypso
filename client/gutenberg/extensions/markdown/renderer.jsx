/** @format */

/**
 * External dependencies
 */
import MarkdownIt from 'markdown-it';
import { __ } from '@wordpress/i18n';
import { RawHTML } from '@wordpress/element';

/**
 * Module variables
 */
const markdownConverter = new MarkdownIt();
const handleLinkClick = event => {
	if ( event.target.nodeName === 'A' ) {
		const hasConfirmed = window.confirm(
			__( 'Are you sure you wish to leave this page?', 'jetpack' )
		);

		if ( ! hasConfirmed ) {
			event.preventDefault();
		}
	}
};

export default ( { className, source = '' } ) => (
	<RawHTML className={ className } onClick={ handleLinkClick }>
		{ source.length ? markdownConverter.render( source ) : '' }
	</RawHTML>
);
