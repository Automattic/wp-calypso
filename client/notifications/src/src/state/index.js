import { applyMiddleware, combineReducers, compose, createStore } from 'redux';

import actionMiddleware from './action-middleware';
import notes from './notes/reducer';
import suggestions from './suggestions/reducer';
import ui from './ui/reducer';

const reducer = combineReducers({
  notes,
  suggestions,
  ui,
});

/** @see https://github.com/zalmoxisus/redux-devtools-extension */
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const withMiddleware = customMiddleware =>
  composeEnhancers(applyMiddleware(actionMiddleware(customMiddleware)))(createStore);

export const init = ({ customEnhancer, customMiddleware = {} } = {}) => {
  const middle = withMiddleware(customMiddleware);
  const create = customEnhancer ? customEnhancer(middle) : middle;

  store = create(reducer, reducer(undefined, { type: '@@INIT' }));

  return store;
};

export let store = init();
