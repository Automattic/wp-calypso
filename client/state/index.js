/**
 * External dependencies
 */
import thunkMiddleware from 'redux-thunk';
import { createStore, applyMiddleware, combineReducers, compose } from 'redux';

/**
 * Internal dependencies
 */
import noticesMiddleware from './notices/middleware';
import postsEditMiddleware from './posts/middleware';
import application from './application/reducer';
import comments from './comments/reducer';
import componentsUsageStats from './components-usage-stats/reducer';
import currentUser from './current-user/reducer';
import documentHead from './document-head/reducer';
import domains from './domains/reducer';
import googleAppsUsers from './google-apps-users/reducer';
import jetpackConnect from './jetpack-connect/reducer';
import jetpackSync from './jetpack-sync/reducer';
import happychat from './happychat/reducer';
import notices from './notices/reducer';
import pageTemplates from './page-templates/reducer';
import plans from './plans/reducer';
import plugins from './plugins/reducer';
import posts from './posts/reducer';
import postTypes from './post-types/reducer';
import preferences from './preferences/reducer';
import preview from './preview/reducer';
import productsList from './products-list/reducer';
import pushNotifications from './push-notifications/reducer';
import purchases from './purchases/reducer';
import reader from './reader/reducer';
import receipts from './receipts/reducer';
import sharing from './sharing/reducer';
import signup from './signup/reducer';
import sites from './sites/reducer';
import siteSettings from './site-settings/reducer';
import stats from './stats/reducer';
import storedCards from './stored-cards/reducer';
import support from './support/reducer';
import terms from './terms/reducer';
import themes from './themes/reducer';
import ui from './ui/reducer';
import users from './users/reducer';
import wordads from './wordads/reducer';

/**
 * Module variables
 */
export const reducer = combineReducers( {
	application,
	comments,
	componentsUsageStats,
	currentUser,
	documentHead,
	domains,
	googleAppsUsers,
	happychat,
	jetpackConnect,
	jetpackSync,
	notices,
	pageTemplates,
	plugins,
	plans,
	preferences,
	preview,
	posts,
	postTypes,
	productsList,
	purchases,
	pushNotifications,
	reader,
	receipts,
	sharing,
	signup,
	sites,
	siteSettings,
	stats,
	storedCards,
	support,
	terms,
	themes,
	ui,
	users,
	wordads
} );

const middleware = [ thunkMiddleware, noticesMiddleware, postsEditMiddleware ];

if ( typeof window === 'object' ) {
	// Browser-specific middlewares
	middleware.push(
		require( './analytics/middleware.js' ).analyticsMiddleware
	);
}

let createStoreWithMiddleware = applyMiddleware.apply( null, middleware );

export function createReduxStore( initialState = {} ) {
	if (
		typeof window === 'object' &&
		window.app &&
		window.app.isDebug &&
		window.devToolsExtension
	) {
		createStoreWithMiddleware = compose( createStoreWithMiddleware, window.devToolsExtension() );
	}
	return createStoreWithMiddleware( createStore )( reducer, initialState );
}
