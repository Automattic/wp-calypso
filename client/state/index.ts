import { isEnabled } from '@automattic/calypso-config';
import { useSelector as reduxUseSelector, useDispatch as reduxUseDispatch } from 'react-redux';
import { createStore, applyMiddleware, compose, Store, StoreEnhancer } from 'redux';
import dynamicMiddlewares from 'redux-dynamic-middlewares';
import thunkMiddleware from 'redux-thunk';
import wpcomApiMiddleware from 'calypso/state/data-layer/wpcom-api-middleware';
import { addReducerEnhancer } from 'calypso/state/utils/add-reducer-enhancer';
import actionLogger from './action-log';
import consoleDispatcher from './console-dispatch';
import initialReducer from './reducer';
import { IAppState, CalypsoDispatch } from './types';

export function createReduxStore(
	initialState: Record< string, unknown >,
	reducer = initialReducer
): Store {
	const isBrowser = typeof window === 'object';
	const isDesktop = isEnabled( 'desktop' );
	const isAudioSupported = typeof window === 'object' && typeof window.Audio === 'function';

	const middlewares = [
		thunkMiddleware,
		// We need the data layer middleware to be used as early
		// as possible, before any side effects.
		// The data layer dispatches actions on network events
		// including success, failure, and progress updates
		// Its way of issuing these is to wrap the originating action
		// with special meta and dispatch it again each time.
		// If another middleware jumps in before the data layer
		// then it could mistakenly trigger on those network
		// responses. Therefore we need to inject the data layer
		// as early as possible into the middleware chain.
		wpcomApiMiddleware,
		isBrowser && require( './happychat/middleware.js' ).default,
		isBrowser && require( './happychat/middleware-calypso.js' ).default,
		dynamicMiddlewares,
		isBrowser && require( './analytics/middleware.js' ).analyticsMiddleware,
		isBrowser && require( './lib/middleware.js' ).default,
		isAudioSupported && require( './audio/middleware.js' ).default,
		isDesktop && require( './desktop/middleware.js' ).default,
	].filter( Boolean );

	const enhancers = [
		addReducerEnhancer,
		isBrowser && window.app && window.app.isDebug && consoleDispatcher,
		applyMiddleware( ...middlewares ),
		isBrowser && window.app && window.app.isDebug && actionLogger,
		isBrowser && window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
	].filter( Boolean );

	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
	// @ts-ignore The store enhancer types don't line up exactly and it's not worth fixing immediately.
	return createStore( reducer, initialState, compose< StoreEnhancer >( ...enhancers ) );
}

export const useDispatch: () => CalypsoDispatch = reduxUseDispatch;

// Basically the same types as TypedUseSelectorHook but with IAppState as the default state type.
export function useSelector< State = IAppState, Selected = unknown >(
	selector: ( state: State ) => Selected,
	equalityFn?: ( a: Selected, b: Selected ) => boolean
): Selected {
	return reduxUseSelector( selector, equalityFn );
}
