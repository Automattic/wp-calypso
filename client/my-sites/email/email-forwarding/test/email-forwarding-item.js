/**
 * External dependencies
 */
import React from 'react';
import renderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import { createReduxStore } from 'calypso/state';
import EmailForwardingItem from '../email-forwarding-item';

describe( 'EmailForwardingItem', () => {
	test( 'it renders EmailForwardingItem correctly', () => {
		const store = createReduxStore();
		const tree = renderer
			.create(
				<EmailForwardingItem
					emailData={ {
						domain: 'foo.com',
						forward_address: 'foo@a8c.com',
						mailbox: 'foo',
					} }
					store={ store }
				/>
			)
			.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
