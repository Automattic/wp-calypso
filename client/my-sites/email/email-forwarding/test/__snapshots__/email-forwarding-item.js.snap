// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailForwardingItem it renders EmailForwardingItem correctly 1`] = `
<li>
  <button
    className="button is-borderless"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="gridicon gridicons-trash"
      height={24}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlinkHref="gridicons.svg#gridicons-trash"
      />
    </svg>
  </button>
  <button
    className="button is-borderless"
    onClick={[Function]}
    title="Resend Verification Email"
    type="button"
  >
    <svg
      className="gridicon gridicons-mail"
      height={24}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlinkHref="gridicons.svg#gridicons-mail"
      />
    </svg>
  </button>
  <span>
    <strong>
      
    </strong>
     
    <em>
      forwards to
    </em>
     
    <strong>
      foo@a8c.com
    </strong>
  </span>
</li>
`;
